app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -166,6 +170,19 @@ def create
     changed_value.nil? ? 'None' : changed_value
   end
 
+  private def push_previous_changes_to_changes(previous_changes, changes)
```

Comment:
I am not a big fan of manipulating (writing to) values that are passed as parameter.
This method should just return a "friendly" version of `previous_changes`, i.e. map it to `UserRoleChange` values and then whoever is using the return value from this method can append it to `changes`.

Reply:
Okay, I've made the method return the changes list and appended it from where it's called.